pkg/ccl/backupccl/restoration_data.go

```diff
@@ -148,3 +164,21 @@ func checkForMigratedData(details jobspb.RestoreDetails, dataToRestore restorati
 
 	return false
 }
+
+// findLatestIntroBySpan finds the latest intro time for the inputted spans.
+// This function assumes that each span's start and end key belong to the same
+// index.
+func findLatestIntroBySpan(
```

Comment:
why do we need this method, I think I'm missing something. Why can't we just operate on the introduced spans mapping that we created initially and in `makeSimpleImportSpans` just check if the span being processed is part of that mapping, and if it is then perform the relevant time filtering. I don't quite get why we map[{tableID,indexID}]:[]ts and then pivot that to []ts that is 1:1 with the passed in spans slice.

Reply:
good q. I initially implemented what you proposed, but chose the current approach for modularity. My high level thinking is: "during makeSimpleImportSpans, we operate with spans. we shouldn't get in the business of decoding keys for table IDs-- that work should happen before we call `restoreWithRetry`. 

On the other hand, maybe we should do what you're suggesting-- this function you highlight isn't strictly necessary. Further, we implicitly do operate on the index level in `makeSimpleImportSpans`, as each span represents an index. If you prefer your approach, I'm happy to change it.

Reply:
I think it simplifies things if we have a single method that creates the mapping and thats our source of truth throughout. We're not paying an additional cost in decoding the key since the number of spans we're passing into `makeSimpleImportSpans` is the same as if we did it individually.